Src/FluentAssertions.Json/JTokenDifferentiator.cs

```diff
@@ -191,27 +198,30 @@ private static Difference FindJObjectDifference(JObject actual, JToken expected,
                 return new Difference(DifferenceKind.OtherName, path);
             }
             
-            return FindFirstDifference(actualProperty.Value, expectedProperty.Value, path, ignoreExtraProperties);
+            return FindFirstDifference(actualProperty.Value, expectedProperty.Value, path, ignoreExtraProperties, config);
         }
 
-        private static Difference FindValueDifference(JValue actualValue, JToken expected, JPath path)
+        private static Difference FindValueDifference(JValue actualValue, JToken expected, JPath path, Func<EquivalencyAssertionOptions<object>, EquivalencyAssertionOptions<object>> config = null)
         {
             if (!(expected is JValue expectedValue))
             {
                 return new Difference(DifferenceKind.OtherType, path, Describe(actualValue.Type), Describe(expected.Type));
             }
             
-            return CompareValues(actualValue, expectedValue, path);
+            return CompareValues(actualValue, expectedValue, path, config);
         }
 
-        private static Difference CompareValues(JValue actual, JValue expected, JPath path)
+        private static Difference CompareValues(JValue actual, JValue expected, JPath path, Func<EquivalencyAssertionOptions<object>, EquivalencyAssertionOptions<object>> config = null)
         {
             if (actual.Type != expected.Type)
             {
                 return new Difference(DifferenceKind.OtherType, path, Describe(actual.Type), Describe(expected.Type));
             }
-            
-            if (!actual.Equals(expected))
+            try
```

Comment:
Instead of using a try/catch, you should be able to create a new `AssertionScope` and check if it has any errors.
See e.g. https://github.com/fluentassertions/fluentassertions/blob/577d555b10d3e393dd3284c1cc3f8ba696def617/Src/FluentAssertions/Primitives/ObjectAssertions.cs#L199-L221